### PR TITLE
Signup worker

### PR DESCRIPTION
**Pull request by jonahbron**

### CODE REVIEW

File: frontend/src/element/signup.ts [hunk 1]
```diff
@@ -1,12 +1,14 @@
 import {
   customElement,
   html,
+  css,
   LitElement,
   TemplateResult,
   internalProperty,
   queryAssignedNodes,
```
Comment (butterflyhug) [inline frontend/src/element/signup.ts:8]:
```suggestion
```

Comment (butterflyhug) [inline frontend/src/element/signup.ts:8]:
```suggestion
```

File: frontend/src/element/signup.ts [hunk 2]
```diff
@@ -27,43 +29,40 @@ import {
...
-  routingNumber!: HTMLInputElement;
+  routingNumber: HTMLInputElement;
   @query('[name="account-number"]')
-  accountNumber!: HTMLInputElement;
+  accountNumber: HTMLInputElement;
   @query('[name="account-holder-name"]')
-  accountHolderName!: HTMLInputElement;
+  accountHolderName: HTMLInputElement;
+  @query('[name="currency"]')
+  currency!: HTMLInputElement;
```
Comment (butterflyhug) [inline frontend/src/element/signup.ts:57]: Nit: is the non-null assertion (`!`) intended here?  I’m not immediately spotting a reason why it’s more important for this element than the others.

File: frontend/src/element/signup.ts [hunk 3]
```diff
@@ -91,96 +94,161 @@ export class Signup extends LitElement {
...
+        >
+          Card
+        </button>
+      </div>
+      ${this.paymentMethod === 'bank' ? this.bankTemplate : this.cardTemplate}
       <button @click=${this.submit}>Submit</button>
     `;
   }
 
+  static get styles() {
```
Comment (butterflyhug) [inline frontend/src/element/signup.ts:199]:
```suggestion
  static get styles(): CSSResult {
```

File: signup-worker/README.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
+# ʕ •́؈•̀) `workers-typescript-template`
```
Comment (butterflyhug) [inline signup-worker/README.md:1]: Should we update this?

File: signup-worker/package.json [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  "scripts": {
+    "build": "webpack",
+    "dev": "NODE_ENV=development npm run build",
+    "format": "prettier --write '**/*.{ts,js,css,json,md}'",
+    "test:clean": "rimraf ./transpiled/src ./transpiled/test",
+    "test": "npm run test:clean && npm run transpile && mocha --require source-map-support/register --recursive transpiled/test",
+    "transpile": "tsc --project ./test"
+  },
+  "author": "author",
+  "license": "MIT OR Apache-2.0",
```
Comment (butterflyhug) [inline signup-worker/package.json:15]: Why the dual license?

File: signup-worker/src/handler.ts [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+export async function handleRequest(request: Request): Promise<Response> {
+  try {
+    const fields = await request.formData();
+    const customerResponse = await stripeClient.createCustomer({
+      source: fields.get('stripe-payment-token') as string,
+    });
+    const customer = await customerResponse.json();
+    const subscriptionResponse = await stripeClient.createSubscription({
+      customer: customer.id,
+      //'pause_collection[behavior]': 'keep_as_draft', // TODO make a second call to set this?
```
Comment (butterflyhug) [inline signup-worker/src/handler.ts:33]: Looks like the TODO is implemented, can we delete this?
```suggestion
```

File: signup-worker/src/handler.ts [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+          currency: (fields.get('currency') as string),
+          product: DUES_PRODUCT_ID,
+          unit_amount: totalCompDollarsToBillingCycleDuesCents(Number(fields.get('total-compensation') as string)),
+          recurring: {
+            interval: 'month',
+          },
+        },
+      }],
+      add_invoice_items: [
+        {price: DUES_SIGNUP_PRICE_ID}
```
Comment (butterflyhug) [inline signup-worker/src/handler.ts:47]:
```suggestion
        {price: DUES_SIGNUP_PRICE_ID},
```

File: signup-worker/src/handler.ts [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+      ],
+    });
+    const subscription = await subscriptionResponse.json();
+    const updateResponse = await stripeClient.updateSubscription(subscription.id, {
+      pause_collection: {
+        behavior: 'keep_as_draft',
+      }
```
Comment (butterflyhug) [inline signup-worker/src/handler.ts:54]:
```suggestion
      },
```

File: signup-worker/src/handler.ts [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+    });
+    return new Response(`Created customer: ${await updateResponse.text()}`, {headers: {'Access-Control-Allow-Origin': '*'}})
+  } catch (e) {
+    console.log(e);
```
Comment (butterflyhug) [inline signup-worker/src/handler.ts:58]: Optional: maybe log at a more prominent level (warning or error)?
```suggestion
    console.warn(e);
```

File: signup-worker/src/index.ts [hunk 10]
```diff
@@ -0,0 +1,5 @@
+import { handleRequest } from './handler'
+
+addEventListener('fetch', (event) => {
+  event.respondWith(handleRequest(event.request))
```
Comment (butterflyhug) [inline signup-worker/src/index.ts:4]:
```suggestion
  event.respondWith(handleRequest(event.request));
```

File: signup-worker/src/handler.ts [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+    return new Response(`Failed: ${e}`, {headers: {'Access-Control-Allow-Origin': '*'}})
```
Comment (butterflyhug) [inline signup-worker/src/handler.ts:59]:
```suggestion
    return new Response(`Failed: ${e}`, {headers: {'Access-Control-Allow-Origin': '*'}});
```

File: frontend/src/element/signup.ts [hunk 12]
```diff
@@ -199,7 +267,7 @@ export class Signup extends LitElement {
       address_state: this.billingState.value,
       address_zip: this.billingZip.value,
       address_country: this.billingCountry.value,
-      currency: 'usd',
+      currency: this.currency.value,
```
Comment (butterflyhug) [inline frontend/src/element/signup.ts:270]: Should we also update `bankAccountToken` to use `this.currency.value` on line 283?

**Review by butterflyhug — APPROVED** [review]
Thanks!

File: frontend/src/element/signup.ts (hunk 2 above)
Comment (jonahbron) [reply frontend/src/element/signup.ts:57]: I'm asserting non-null on elements that I know are always present.  Some are behind conditions.

File: signup-worker/package.json (hunk 5 above)
Comment (jonahbron) [reply signup-worker/package.json:15]: A lot of this is pulled in from a template.  Will change.

**Review by butterflyhug — APPROVED** [review]

**Review by justaliu — APPROVED** [review]

**Pull request merged**